frontend-react/src/components/MessageTracker/MessageTracker.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from "react";
+import { Label, Button, TextInput } from "@trussworks/react-uswds";
+
+import Spinner from "../Spinner";
+import Table, { TableConfig } from "../../components/Table/Table";
+
+const MOCK_MESSAGE_SENDER_DATA = [
+    {
+        messageId: "12-234567",
+        sender: "somebody 1",
+        submittedDate: "09/28/2022",
+        reportId: "29038fca-e521-4af8-82ac-6b9fafd0fd58",
+    },
+    {
+        messageId: "12-234567",
+        sender: "somebody 2",
+        submittedDate: "09/29/2022",
+        reportId: "86c4c66f-3d99-4845-8bea-111210c05e63",
+    },
+    {
+        messageId: "12-234567",
+        sender: "somebody 3",
+        submittedDate: "09/30/2022",
+        reportId: "26a37945-ed12-4578-b4f6-203e8b9d62ce",
+    },
+];
+
+// TODO: move this interface into the resources directory
+export interface MessageListResource {
+    messageId: string;
+    sender: string | undefined;
+    submittedDate: string | undefined;
+    reportId: string;
+}
+
+interface MessageListTableContentProps {
+    isLoading: boolean;
+    messagesData: MessageListResource[];
+}
+
+const MessageTrackerTableContent: React.FC<MessageListTableContentProps> = ({
+    isLoading,
+    messagesData,
+}) => {
+    const tableConfig: TableConfig = {
+        columns: [
+            {
+                dataAttr: "messageId",
+                columnHeader: "Message Id",
+                feature: {
+                    link: true,
+                    linkBasePath: "/message-details/",
+                },
+            },
+            {
+                dataAttr: "sender",
+                columnHeader: "Sender",
+            },
+            {
+                dataAttr: "submittedDate",
+                columnHeader: "Date/time submitted",
+                transform: (s: string) => {
+                    return new Date(s).toLocaleString();
+                },
+            },
+            {
+                dataAttr: "reportId",
+                columnHeader: "Incoming Report Id",
+                feature: {
+                    link: true,
+                    linkBasePath: "/report-details?reportId=",
+                },
+            },
+        ],
+        rows: messagesData || [],
+    };
+
+    if (isLoading) return <Spinner />;
+
+    return (
+        <>
+            {messagesData.length > 0 && (
+                <Table
+                    title=""
+                    classes={"rs-no-padding margin-top-5"}
+                    config={tableConfig}
+                />
+            )}
+        </>
+    );
+};
+
+// Main component.
+export function MessageTracker() {
+    const [searchFilter, setSearchFilter] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    const [messagesData, setMessagesData] = useState<MessageListResource[]>([]);
+
+    const searchMessageId = async () => {
+        setIsLoading(true);
+
+        // TODO: make API call here to get data
```

Comment:
Is this for prototyping purposes, or are we going to need to fetch different from how we do with react-query?